scipy/ndimage/src/ni_fourier.c

```diff
@@ -225,12 +225,10 @@ int NI_FourierFilter(PyArrayObject *input, PyArrayObject* parameter_array,
         params[kk] = NULL;
     }
     for (kk = 0; kk < PyArray_NDIM(input); kk++) {
-        if (PyArray_DIM(input, kk) > 1) {
-            params[kk] = malloc(PyArray_DIM(input, kk) * sizeof(double));
```

Comment:
I think this might create new issues with length-0 axes. Specifically, `_NI_UNIFORM` on line 270 has:
```c
if (params[hh]) {
	params[hh][0] = 1.0;
    // etc...
}
```
which is invalid if `params[hh]` is a non-null pointer to a 0-sized array.

Reply:
Yes, you're right. I didn't notice as there are no tests for length-0 dimensions, perhaps not surprisingly as there were no tests for length-1 dimensions.

So, should length-0 dimensions be supported? In general this seems like a bad idea but I am new around here.

Reply:
I would say it's better to support empty dimensions unless there's a good reason not to. In this case, the result should just be an empty array with the same shape as the input. So long as the parameter initialization doesn't crash it should be fine.

Also note places where the code already checks for `> 0`, so again it's making an effort to support it.